smk/gene_abundance.smk

```diff
@@ -662,6 +716,76 @@ rule merge_msamtools_profiles:
         ) >& {log}
         """
 
+rule add_annotation_to_genome_profiles:
+    input:
+        profile  = "{wd}/{omics}/9-mapping-profiles/{minto_mode}/genome_abundances.p{identity}.relabund.prop.tsv",
+        locusmap = "{wd}/DB/{minto_mode}/1-prokka/locus_id_list.txt",
+        taxonomy = "{wd}/DB/{minto_mode}/3-taxonomy/taxonomy.{taxonomy}.{db_version}.tsv"
+    output:
+        mag_profile = "{wd}/output/9-mapping-profiles/{minto_mode}/{omics}.{taxonomy}.{db_version}.p{identity}.tsv",
+        sp_profile  = "{wd}/output/9-mapping-profiles/{minto_mode}/{omics}.{taxonomy}.{db_version}.p{identity}.species.tsv"
+    log:
+        "{wd}/logs/{omics}/9-mapping-profiles/{minto_mode}/{omics}.{taxonomy}.{db_version}.p{identity}.log"
+    wildcard_constraints:
+        taxonomy='gtdb|phylophlan',
+        identity='[0-9]+'
+    resources:
+        mem=10
+    conda:
+        config["minto_dir"]+"/envs/r_pkgs.yml"
+    shell:
+        """
+        time (
+            R --vanilla --silent --no-echo >> {output} <<___EOF___
+
+library(data.table)
+
+# Read tables
+profile  = fread('{input.profile}',  header = TRUE, sep = "\\t")
+locusmap = fread('{input.locusmap}', header = TRUE, sep = "\\t")
+taxonomy = fread('{input.taxonomy}', header = TRUE, sep = "\\t")
```

Comment:
I believe, the taxonomy fread needs a `fill = TRUE` to read the lines where no species, genus, etc. are printed (fewer columns than in header).

Reply:
It's filled with `Unknown` later - about 10 lines later.
```
[, lapply(.SD, function(x) ifelse(is.na(x), 'Unknown', x))]
```

I have now added comments to make this clear:	3dbecf7

Reply:
In my testing, those cells were empty, not NA, and thus they stay empty. (And there is a warning from fread on the number of columns.)

Reply:
Strange! It used to work without using `fill` and checking only for `NA` during my tests. All the output I shared for your philosophical question was without that fix, but showed the right output. Now when I ran it again, it failed with same error like you. I must have done something wrong during the tests. Fixed now: bcce174

Reply:
I think it's due to the new way the phylophlan taxonomy is prepared: the initial commit would have added all columns even if empty, and the second one does not produce those.